state/src/interface/peer_index.rs

```diff
@@ -125,70 +138,76 @@ impl<N: NetworkEssential> StateHandle<N> {
     /// Add a batch of peers to the index
     pub async fn add_peer_batch(&self, peers: Vec<PeerInfo>) -> Result<(), StateError> {
         // Index each peer
-        let tx = self.db.new_write_tx()?;
-        for mut peer in peers.into_iter() {
-            // Parse the peer info and mark a successful heartbeat
-            peer.successful_heartbeat();
-
-            // Do not index the peer if the given address is not dialable
-            if !peer.is_dialable(self.allow_local) {
-                continue;
-            }
-
-            // Add the peer to the store
-            tx.write_peer(&peer)?;
-            tx.add_to_cluster(&peer.peer_id, &peer.cluster_id)?;
-
-            // If the peer belongs in the same cluster, add it to the raft group
-            let my_cluster_id = tx.get_cluster_id()?;
-            if peer.cluster_id == my_cluster_id {
-                let raft_id = PeerIdTranslationMap::get_raft_id(&peer.peer_id);
-                self.raft.add_learner(raft_id).await?;
+        let this = self.clone();
+        self.with_write_tx(move |tx| {
+            for mut peer in peers.into_iter() {
+                // Parse the peer info and mark a successful heartbeat
+                peer.successful_heartbeat();
+
+                // Do not index the peer if the given address is not dialable
+                if !peer.is_dialable(this.allow_local) {
+                    continue;
+                }
+
+                // Add the peer to the store
+                tx.write_peer(&peer)?;
+                tx.add_to_cluster(&peer.peer_id, &peer.cluster_id)?;
+
+                // If the peer belongs in the same cluster, add it to the raft group
+                let my_cluster_id = tx.get_cluster_id()?;
+                if peer.cluster_id == my_cluster_id {
+                    let raft_id = PeerIdTranslationMap::get_raft_id(&peer.peer_id);
+                    block_current(this.raft.add_learner(raft_id))?;
```

Comment:
unable to find `block_current` in this PR, if the idea is that it blocks the current runtime until this task is complete, is that so that any other concurrent proposals are sent to the right set of peers?